src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/PipelineTests.cs

```diff
@@ -59,20 +60,21 @@ protected override bool ShouldRetry(HttpPipelineMessage message, int retry, out
 
         class NullPipelineContext : HttpPipelineRequest
         {
-            public override void SetRequestLine(HttpVerb method, Uri uri)
+            public override void AddHeader(HttpHeader header)
             {
             }
 
-            public override void AddHeader(HttpHeader header)
+            public override bool TryGetHeader(string name, out string value)
             {
+                value = null;
+                return false;
             }
 
-            public override void SetContent(HttpPipelineRequestContent content)
+            public override IEnumerable<HttpHeader> GetHeaders()
```

Comment:
Also, we need to decide on the contract of this method. Should the header value have all concatenated values or it can return multiple headers with the same name for each individual header value.

Reply:
I say let it return concatenated value. I doubt policies that specifically care about uncontatenated values would iterate over all the headers to get them. We can have a separate method to get unconcatenated values `TryGetHeader(string name, out IEnumerable<string> values)` for the ones that do care.

Reply:
Personally, I like the single-header-with-concatinated-values approach.